forth.asm

```diff
@@ -203,6 +203,28 @@ dict_entry tonumber, '>NUMBER', 0x0 ;( str -- n)
     push    ebx
     jmp     next
 
+interpret_if:
+    ; if stack nozero, advance instruction pointer
+    pop     eax
+    test    eax, eax
+    jnz     .true
+    add     esi, 0x8
+    jmp     next
+.true:
+    jmp     next
+
+
+dict_entry interpret, 'INTERPRET-WORD', 0x0 ; ( str -- )
+    call    docolon
+    dd      blank
+    dd      cword
+    dd      find
+    dd      interpret_if
```

Comment:
I think now it snaps back to me, that I was to "lazy" to distinguish between words used in assembly and words executed by the interpreter. Or is there something else I can do?